# Description: let the logout key do what it's supposed to do
# Ubuntu: https://launchpad.net/bugs/961501
# Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=671979
---
# plugins/media-keys/gsd-media-keys-manager.c |    8 +++++++-
# 1 file changed, 7 insertions(+), 1 deletion(-)
#
--- a/plugins/media-keys/gsd-media-keys-manager.c
+++ b/plugins/media-keys/gsd-media-keys-manager.c
@@ -940,6 +940,12 @@ gnome_session_shutdown (GsdMediaKeysMana
 }
 
 static void
+do_logout_action (GsdMediaKeysManager *manager)
+{
+        execute (manager, "gnome-session-quit --logout", FALSE);
+}
+
+static void
 do_eject_action_cb (GDrive              *drive,
                     GAsyncResult        *res,
                     GsdMediaKeysManager *manager)
@@ -2335,7 +2341,7 @@ do_action (GsdMediaKeysManager *manager,
                                  SOUND_ACTION_FLAG_IS_OUTPUT | SOUND_ACTION_FLAG_IS_PRECISE);
                 break;
         case LOGOUT_KEY:
-                gnome_session_shutdown (manager);
+                do_logout_action (manager);
                 break;
         case EJECT_KEY:
                 do_eject_action (manager);
